identity/src/keypair.rs

```diff
@@ -342,6 +342,26 @@ impl Keypair {
             KeyPairInner::Ecdsa(_) => KeyType::Ecdsa,
         }
     }
+
+    /// Return the secret key of the [`Keypair`] if it has one.
+    pub fn secret(&self) -> Option<[u8; 32]> {
```

Comment:
Instead of returning the secret key directly, I'd like to return something that is derived from it and the user has to pass a domain seperater. We can use the `hkdf` crate for that to not depend on ring here.

For the stateless reset key, we can just apply another round of hmac to fulfill the API requirement of quinn.

I think this is to rustls always depending on ring but that has recently changed so we might be able to move away from that: https://github.com/rustls/rustls/pull/1405

Reply:
yeah makes total sense, Thomas had already mentioned it when defined the api, sorry I missed it. Updated ptal :)